### PR TITLE
Fix for the radio button not working with onChange

### PR DESCRIPTION
**Pull request by cedmax**
As suggested in the comment to the merged PR (ref: https://github.com/developit/preact-compat/pull/247) the event normalisation has broken the support for onChange on radio buttons. This is a potential fix

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -337,7 +337,7 @@ function applyEventNormalization({ nodeName, attributes }) {
 	}
 	if (props.onchange) {
 		nodeName = nodeName.toLowerCase();
-		let attr = nodeName==='input' && String(attributes.type).toLowerCase()==='checkbox' ? 'onclick' : 'oninput',
+		let attr = nodeName==='input' && ['checkbox', 'radio'].indexOf(String(attributes.type).toLowerCase()) > -1 ? 'onclick' : 'oninput',
```
Comment (developit) [inline src/index.js]: Looks great! Any chance we could shorten this to the technique used [here](https://github.com/developit/preact/blob/master/src/linked-state.js#L16)?

```js
let attr = nodeName==='input' && attributes.type.match(/^che|rad/i) ? 'onclick' : 'oninput',
```

Comment (cedmax) [inline src/index.js]: yep, it makes sense, I will make the change

Comment (cedmax) [inline src/index.js]: just one thing: do you want me to lose the String(...).toLowerCase(): it was in the original, that's why I kept it

Comment (cedmax) [inline src/index.js]: The PR is updated without it anyway, if you need it back let me know

Comment (developit) [inline src/index.js]: We might need the String() wrap, the regex handles case insensitivity.

Comment (developit) [inline src/index.js]: Good catch on that, it would have bailed for `<input />` without a type.  Let's switch to `.test()` since that doesn't care if the argument is a string:

```js
let attr = nodeName==='input' && /^che|rad/i.test(attributes.type) ? 'onclick' : 'oninput',
```

**Comment by developit** [comment]
(left this inline but it got collapsed by github haha)

Good catch on that, it would have bailed for `<input />` without a type.  Let's switch to `.test()` since that doesn't care if the argument is a string:

```js
let attr = nodeName==='input' && /^che|rad/i.test(attributes.type) ? 'onclick' : 'oninput',
```

we're good to merge once that's in place :)

I might add a few tests for posterity post-merge.

**Comment by cedmax** [comment]
The PR is updated with the changes, thanks for the speedy process

**Comment by developit** [comment]
thanks for the speedy fix!

**Comment by ustccjw** [comment]
@developit why delete onclick for radio now?
